### PR TITLE
[MKL-DNN DOC] Added mkl-dnn design doc on inter-op Tensors conversion

### PR DESCRIPTION
**Pull request by jczaja**
This is first in a series of documents describing architecture of mkl-dnn integration. It is meant to be useful for PaddlePaddle developers and mkl-dnn integration maintainers in particular.

Rendered(suitable for reading) version of document can be seen at:
https://github.com/jczaja/Paddle/blob/prv-mkldnn-doc-data-transformation/paddle/fluid/operators/mkldnn/doc/data_transformation/data_transform.md

@grygielski , @bingyanghuang  Please review

### CODE REVIEW

**Review by grygielski — APPROVED** [review]
LGTM

File: paddle/fluid/operators/mkldnn/doc/data_transformation/data_transform.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+When fluid is using MKL-DNN engine to execute program, not all operators are having mkl-dnn kernels and some of operators are executed by CPU. MKL-DNN kernels of operators expect input Tensors to be provided in MKL-DNN layout , while Paddle(CPU) kernels are expecting input Tensors to be of Paddle layout.
+
+We can distinguish following scenarios(presented below on the picture):
+* Paddle(CPU) kernel is followed by MKL-DNN kernel
+* MKL-DNN kernel is followed by Paddle(CPU) kernel
+* MKL-DNN kernel is followed by fetch operator
+
+
+![](images/data_transform.svg)
```
Comment (luotao1) [inline paddle/fluid/operators/mkldnn/doc/data_transformation/data_transform.md:11]: https://github.com/PaddlePaddle/Paddle/blob/63bca64fa29a1b2e719ce47097c9fad57f050d6d/paddle/fluid/operators/mkldnn/doc/data_transformation/data_transform.md I could not see the picture here.

Comment (jczaja) [reply paddle/fluid/operators/mkldnn/doc/data_transformation/data_transform.md:11]: @luotao1 I have used SVG as it rendering it should be better quality than PNG. Let me know if there is a need to switch to PNG images

Comment (luotao1) [reply paddle/fluid/operators/mkldnn/doc/data_transformation/data_transform.md:11]: No, this PR LGTM!

**Comment by jczaja** [comment]
@luotao1 Could you please continue your review?

**Comment by bingyanghuang** [comment]
@jczaja As we talked with Luotao in today's weekly meeting, we both think that this document would be better placed in FluidDoc as a design doc not published formerly. An.d for the following architectural design docs , you can also put them in FluidDoc. So please close this PR and reopen your PR in FluidDoc. Sorry for confusion

**Comment by jczaja** [comment]
@bingyanghuang Ok, I have a question on how make FluidDoc PR not formally published. Should I skip adding RST files (Index files needed so that MD is part of generated documentation) ?

**Comment by luotao1** [comment]
All the design docs in FluidDoc are not published on the https://www.paddlepaddle.org.cn/documentation/docs/en/advanced_usage/design_idea/fluid_design_idea_en.html# @jczaja So you can add index files.

**Comment by jczaja** [comment]
Closing due to migration of this design doc into FluidDoc repo

**Pull request closed without merging**